secboot/secboot_tpm.go

```diff
@@ -379,23 +381,33 @@ func ResealKeys(params *ResealKeysParams) error {
 		return fmt.Errorf("cannot read the policy auth key file: %v", err)
 	}
 
-	return sbUpdateKeyPCRProtectionPolicyMultiple(tpm, params.KeyFiles, authKey, pcrProfile)
+	numSealedKeyObjects := len(params.KeyFiles)
+	sealedKeyObjects := make([]*sb_tpm2.SealedKeyObject, 0, numSealedKeyObjects)
+	for _, keyfile := range params.KeyFiles {
+		sealedKeyObject, err := sb_tpm2.ReadSealedKeyObject(keyfile)
+		if err != nil {
+			return err
+		}
+		sealedKeyObjects = append(sealedKeyObjects, sealedKeyObject)
+	}
+
+	return sbUpdateKeyPCRProtectionPolicyMultiple(tpm, sealedKeyObjects, authKey, pcrProfile)
```

Comment:
@anonymouse64 I believe this API change is breaking unit tests, as they are not expecting an extra call to `ReadSealedKeyObject()` which is not mocked and throws real "cannot read file" instead of being mocked.

Reply:
probably what you need to do is add to this package:

```go
var sb_tpm2ReadSealedKeyObject = sb_tpm2.ReadSealedKeyObject
```

then here use `sb_tpm2ReadSealedKeyObject`

then in export_test.go add 

```go
func MockReadSealedKeyObject(func f(string) (obj, error)) (restore func()) {
    old := sb_tpm2ReadSealedKeyObject
    sb_tpm2ReadSealedKeyObject = f
    return func() {
        sb_tpm2ReadSealedKeyObject = old
    }
}
```

then all the unit tests which call ResealKeys will be updated to call MockReadSealedKeyObject() with some function which checks/tracks that the calls to sb_tpm2ReadSealedKeyObject are as expected.

Does that make sense?